aiida_common_workflows/workflows/relax/vasp/generator.py

```diff
@@ -0,0 +1,145 @@
+
+        try:
+            potential_mapping = settings['potential_mapping']
+        except KeyError:
+            potential_mapping = 'DEFAULT_PBE'
+
+        process_class = CommonVASPRelaxWorkChain._process_class  # pylint: disable=protected-access
+
+        builder = CommonVASPRelaxWorkChain.get_builder()
+
+        # Set structure
+        builder.structure = structure
+
+        # Set options
+        inputs.options = DataFactory('dict')(dict=options)
+
+        # Set settings
+        # Make sure we add forces and stress for the parser
+        settings.update({'parser_settings': {'add_forces': True, 'add_stress': True}})
+        inputs.settings = DataFactory('dict')(dict=settings)
+
+        # Set workchain related inputs, in this case, give more explicit output to report
+        inputs.verbose = DataFactory('bool')(True)
+
+        # Here we set the protocols fast, moderate and precise. These currently have no formal meaning.
+        # After a while these will be set in the VASP workchain entrypoints.
+        # First a few general on the safe side parameters
+        parameters = {'ismear': 0,
+                      'sigma': 0.2}
+        builder.relax = AttributeDict()
+        builder.converge = AttributeDict()
+        builder.relax.perform = DataFactory('bool')(True)
+        builder.relax.algo = DataFactory('str')('cg')
+
+        # Then some specifics. We rely on the convergence tests in order to have some
+        # common ground on the precision. Here, we test the total energy convergence to within
+        # converge.cutoff_value (100, 10 and 5 meV, respectively)
+        # This is only preliminary to illustrate the concept. In the end we also want to check how the
+        # forces converge on a relaxation event (e.g. displacement of atoms, change of volume etc.).
+        # The convergence workchain supports this, but in order to keep this simple, it is not implemented here.
+        # Also, in the end we would not be calling the convergence workchain, but some more general workchain that handles
+        # parameter suggestions, at least in a subdomain (say, periodic plane wave codes).
+        if protocol == 'fast':
+            parameters.update({'sigma': 0.2,
+                               'algo': 'V'})
+            builder.converge.cutoff_value = DataFactory('float')(0.1)
+            builder.relax.algo = DataFactory('str')('cg')
+            builder.relax.force_cutoff = DataFactory('float')(0.01)
+
+        if protocol == 'moderate':
+            parameters.update({'sigma': 0.1,
+                               'algo': 'F'})
+            builder.converge.cutoff_value =  DataFactory('float')(0.01)
+            builder.relax.algo = DataFactory('str')('cg')
+
+        if protocol == 'precise':
+            parameters.update({'sigma': 0.05,
+                               'algo': 'N'})
+            builder.converge.cutoff_value = DataFactory('float')(0.005)
+            builder.relax.algo = DataFactory('str')('rd')
```

Comment:
Ah, I see now that you *do* implement protocols in some way. Just need to update `_protocols` then

Reply:
Yes, there is now also a `protocol.yml` file along similar lines as for the QE.